### PR TITLE
Replace CentOS content with Oracle Linux content when versions are identical 

### LINKED ISSUE #2: Replace identical RPMs for completeness
When switching from a recently patched CentOS 6 or 7 the converter will leave CentOS built RPMs on the system if they match the Oracle Linux version. This causes no technical issue, when Oracle Linux releases new versions a system will be updated to it but we shouldn't leave the old distributions content in place. For completeness we should `yum reinstall` all CentOS RPMs with those from Oracle Linux.

This has not been seen on CentOS8.

### PR DESCRIPTION
**Pull request by mark-au**
Reinstall any CentOS RPMs with those built by Oracle. 

Resolves #2 and #3.

### CODE REVIEW

File: centos2ol.sh [hunk 1]
```diff
@@ -486,6 +486,23 @@ case "$os_version" in
...
 esac
 
+echo "Testing for remaining CentOS RPMs"
+# If CentOS and Oracle Linux have identically versioned RPMs then those RPMs are left unchanged.
+#  This should have no technical impact but for completeness, reinstall these RPMs
+#  so there is no accidental cross pollination.
+mapfile -t list_of_centos_rpms < <(rpm -qa --qf "%{NAME}-%{VERSION}-%{RELEASE} %{VENDOR}\n" | grep CentOS | awk '{print $1}')
+if [[ -n "${list_of_centos_rpms[*]}" ]]; then
+    echo "Reinstalling RPMs: ${list_of_centos_rpms[*]}"
+    yum -y reinstall "${list_of_centos_rpms[@]}"
```
Comment (mark-au) [inline centos2ol.sh:496]: Do you think this is too aggressive @Djelibeybi ?
I'm not sure what else CentOS sign and unsure how much to soften the blow. 
* We could limit it to reinstalling just hose in `ol*` repos but that means downloading the metadata of all repos (200+MB today on OL7)
* We could limit it to reinstalling just those in `ol?_latest` repos but that may be too narrow

Comment (Djelibeybi) [reply centos2ol.sh:496]: Why not make it someone else's problem and enable it via a command-line parameter, e.g. `--replace-all`?

Comment (mark-au) [reply centos2ol.sh:496]: Good idea, I'll default to just `ol?_latest` repo unless that argument is passed

Comment (Djelibeybi) [reply centos2ol.sh:496]: Then document that this is useful/necessary if user wants to remove the CentOS GPG key. But I would also point out that it is not required to be eligible for support from Oracle.

Comment (Djelibeybi) [reply centos2ol.sh:496]: Honestly, I'd just stop with the distro-sync unless this is passed. It adds a lot of download and time to the process.

Comment (mark-au) [reply centos2ol.sh:496]: Sure, in that case I won't print the non-Oracle RPMs if this option isn't selected as it loses value.

Comment (Djelibeybi) [reply centos2ol.sh:496]: Agreed.

**Comment by mark-au** [comment]
Please feel free to reword within the branch.

**Review by Djelibeybi — APPROVED** [review]

**Pull request merged**